core/resource_monitor.py

```
@@ -0,0 +1,369 @@
+                cpu_percent=cpu_percent,
+                memory_percent=memory_percent,
+                memory_available_mb=memory_available_mb,
+                disk_usage_percent=disk_usage_percent,
+                process_memory_mb=process_memory_mb,
+                timestamp=datetime.now()
+            )
+            
+            # Add to history
+            self._add_to_history(metrics)
+            
+            return metrics
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)